processing/src/main/java/org/apache/druid/query/FrameBasedInlineDataSource.java

```diff
@@ -81,18 +82,22 @@ public RowSignature getRowSignature()
 
   public Sequence<Object[]> getRowsAsSequence()
   {
-
+    final Closer closer = Closer.create();
     final Sequence<Cursor> cursorSequence =
         Sequences.simple(frames)
-                 .flatMap(
+                 .map(
```

Comment:
IMO, it's better closeable management to use a `flatMap` with single-cursor sequences that individually do `withBaggage(maker)`. That way the makers get closed as soon as we're done with them, instead of all at the end.

It doesn't really matter right now for frames, but I'm thinking in case it does matter in the future.

Reply:
changed, though might be nice to make some special `Sequence` for a singleton so we don't have to make a singleton list and do the iterator maker and all that junk, but I don't feel like doing that in this PR.